### ISSUE TITLE
Type annotations and implementation mismatch

### ISSUE THREAD

**Issue by nickgashkov** [aio-libs/yarl#540]
🐞 **Describe the bug**
Judging by `yarl.URL` type annotations, it is required to pass only integers or strings as query string values and lists are prohibited. But, actually, `yarl` correctly handles the provided list of integers or strings as a query string value.

💡 **To Reproduce**
The mismatch can be observed on the following snippet of code:
```python
import yarl
print(yarl.URL("http://localhost:8000").with_query(params=[1, 2, 3]).human_repr())
```

1. Run the given code snippet and observe correct handling of a list of params:
```console
$ python test.py
http://localhost:8000/?params=1&params=2&params=3
```
2. Run the `mypy` against the given code snippet and observe `mypy`s complaints:
```console
$ mypy test.py
test.py:6: error: No overload variant of "with_query" of "URL" matches argument type "List[int]"
test.py:6: note: Possible overload variant:
test.py:6: note:     def with_query(self, **kwargs: Union[str, int]) -> URL
test.py:6: note:     <1 more non-matching overload not shown>
Found 1 error in 1 file (checked 1 source file)
```

💡 **Expected behavior**
No type errors when providing list of query parameters.


📋 **Logs/tracebacks**
—

📋 **Your version of the Python**
<!-- Attach your version of the Python. -->
```console
$ python --version
Python 3.8.6
```

📋 **Your version of the aiohttp/yarl/multidict distributions**
<!-- Attach your version of the distributions in the code blocks below. -->
```console
$ python -m pip show aiohttp                                                                                                                                                                                                                                                                                              0
Name: aiohttp
Version: 3.6.3
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: Nikolay Kim
Author-email: fafhrd91@gmail.com
License: Apache 2
Location: ...
Requires: multidict, yarl, attrs, async-timeout, chardet
Required-by: test
```
```console
$ python -m pip show multidict
Name: multidict
Version: 4.7.6
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: ...
Requires:
Required-by: yarl, aiohttp
```
```console
$ python -m pip show yarl
Name: yarl
Version: 1.5.1
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl/
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: ...
Requires: idna, multidict
Required-by: test, aiohttp
```

📋 **Additional context**
—

**Comment by asvetlov** [comment]
Pull request is welcome!

**Comment by nickgashkov** [comment]
NVM, as it turns out, the fix has already landed on `master` (#529) :) It just hasn't been pushed to PyPI yet.

Closing this as a duplicate of #528.
